lib/resty/etcd.lua

```diff
@@ -63,8 +63,8 @@ function _M.new(opts)
     end
 
     opts.api_prefix = "/v2"
-    local serializer = opts.serializer or "json"
-    opts.serializer = require_package("resty.etcd.serializers.", serializer, "json")
+    local serializer = typeof.string(opts.serializer) and opts.serializer  or "json"
+    opts.serializer = require_package("resty.etcd.serializers." .. serializer, "resty.etcd.serializers.json")
```

Comment:
please take a look at this code, it is simpler and better performance:

```lua
local function require_package(serializer_name)
    if serializer_name then
        local ok, module = pcall(require, "resty.etcd.serializers." .. serializer_name)
        if ok then
            return module
        end
    end

    return require("resty.etcd.serializers.json")
end

...
local serializer_name = typeof.string(opts.serializer) and opts.serializer
opts.serializer = require_package(serializer_name)

Reply:
yep, and I think a function named `require_serializer` will better